### PR TITLE
Deprecate UiTheme to make its removal smooth

### PR DESCRIPTION
**Pull request by DavDo**
**Jira issue:**
https://glia.atlassian.net/browse/MOB-3421

**What was solved?**

**Release notes:**

 - [ ] Feature
 - [x] Ignore
 - [ ] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

**Additional info:**

 - [ ] Is the feature sufficiently tested? All tests fixed? Necessary unit, acceptance, snapshots added? Check that at least new public classes & methods are covered with unit tests
 - [ ] Did you add logging beneficial for troubleshooting of customer issues?
 - [ ] **Did you add new logging?** We would like the logging between platforms to be similar. Refer to [**Logging from Android SDKs** → **Things to consider for newly added logs**](https://glia.atlassian.net/wiki/spaces/ENG/pages/3568861468/Logging+from+Android+SDKs#Things-to-consider-for-newly-added-logs) in Confluence for more information.

**Screenshots:**

### CODE REVIEW

**Review by andrews-moc — COMMENTED** [review]
I think we need to add `@deprecated` with a comment and `@Deprecated` for each element. Like it was discussed recently [here](https://github.com/salemove/android-sdk/pull/730#discussion_r1662294725).

**Comment by AndriiHorishniiMOC** [comment]
Icons customization is still available only on UiTheme.

**Comment by DavDo** [comment]
> Icons customization is still available only on UiTheme.

Yes, but with resource id's only, so I think it would still be possible with XML arguments.

**Comment by DavDo** [comment]
> I think we need to add `@deprecated` with a comment and `@Deprecated` for each element. Like it was discussed recently [here](https://github.com/salemove/android-sdk/pull/730#discussion_r1662294725).



> I think we need to add `@deprecated` with a comment and `@Deprecated` for each element. Like it was discussed recently [here](https://github.com/salemove/android-sdk/pull/730#discussion_r1662294725).

I think this applies to Java only. 
The @Deprecated annotation in Kotlin supports a message parameter, and optionally, replaceWith parameter to suggest a replacement. And in addition to this, Kotlin uses `KDoc` for documentation that doesn't have a `@deprecated` tag.

**Review by AndriiHorishniiMOC — APPROVED** [review]

**Review by gugalo — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/GliaWidgets.java [hunk 1]
```diff
@@ -39,7 +39,12 @@ public class GliaWidgets {
     /**
      * Use with {@link android.os.Bundle} to pass in a {@link UiTheme} as a navigation argument when
      * navigating to {@link com.glia.widgets.chat.ChatActivity}
+     *
+     * @deprecated While UiTheme can still be used for UI customization,
+     * we strongly encourage adopting remote configurations {@link GliaWidgetsConfig.Builder#setUiJsonRemoteConfig(String)}.
+     * This newer approach is more versatile and better suited for future development.
```
Comment (andrews-moc) [inline widgetssdk/src/main/java/com/glia/widgets/GliaWidgets.java:45]: NIT (I'm not insisting): This newer approach => Remote configurations approach.

**Review by andrews-moc — APPROVED** [review]
Does it make sense to ask tech writes to review the text? I don't insist, it's up to you.

File: widgetssdk/src/main/java/com/glia/widgets/GliaWidgets.java [hunk 2]
```diff
@@ -39,7 +39,12 @@ public class GliaWidgets {
     /**
      * Use with {@link android.os.Bundle} to pass in a {@link UiTheme} as a navigation argument when
      * navigating to {@link com.glia.widgets.chat.ChatActivity}
+     *
+     * @deprecated While UiTheme can still be used for UI customization,
+     * we strongly encourage adopting remote configurations {@link GliaWidgetsConfig.Builder#setUiJsonRemoteConfig(String)}.
+     * Remote configurations approach is more versatile and better suited for future development.
```
Comment (1toomas) [inline widgetssdk/src/main/java/com/glia/widgets/GliaWidgets.java:45]: Please update the same sentence everywhere.
(Seems that this is again one of such cases (previously discussed [here](https://github.com/salemove/android-sdk/pull/730#discussion_r1660855313)) that the generator does not render for the [SDK docs](http://android-sdk-docs.glia.com.s3-website-us-east-1.amazonaws.com/widgets/javadoc/com/glia/widgets/GliaWidgets.html).)
```suggestion
     * The remote configurations approach is more versatile and better suited for future development.
```

**Review by 1toomas — APPROVED** [review]
One tiny suggestion.

File: widgetssdk/src/main/java/com/glia/widgets/GliaWidgets.java (hunk 2 above)
Comment (DavDo) [reply widgetssdk/src/main/java/com/glia/widgets/GliaWidgets.java]: We can migrate to an `HTML` type of documentation, that is more Kotlin-friendly and has a way better UI and user experience.

image
image

**Comment by DavDo** [comment]
!squash

**Pull request merged**